### PR TITLE
InferenceDataset for custom directory

### PR DESCRIPTION
**Pull request by kvantricht**

### CODE REVIEW

**Comment by kvantricht** [comment]
@gabrieltseng is it ok for you to do it like this? I'd like to centralize as much as possible the code for this dataset but still be able to construct it from different directories. Later on, we might to expand class methods to with things like `.from_files()` etc, to make it more generic.

**Comment by gabrieltseng** [comment]
This works for me. However, the best way to expose this would be to update `train.py` to have a `spatial_inference_dir` option. 

What exactly is the scenario we are imagining here? Would we have `.tif` files distributed over many folders? I assume we would want to generate spatial predictions for all of them?

File: presto/dataset.py [hunk 1]
```diff
@@ -216,8 +216,8 @@ class WorldCerealInferenceDataset(Dataset):
         "temperature-mean": "temperature_2m",
     }
 
-    def __init__(self):
-        self.path_to_files = data_dir / "inference_areas"
+    def __init__(self, input_dir: Optional[Union[str, Path]] = None):
+        self.path_to_files = data_dir / "inference_areas" if input_dir is None else Path(input_dir)
```
Comment (gabrieltseng) [inline presto/dataset.py:220]: Instead of making `input_dir` optionally `None`, can you just set the default argument to be `data_dir`?

**Comment by kvantricht** [comment]
> This works for me. However, the best way to expose this would be to update `train.py` to have a `spatial_inference_dir` option.
> 
> What exactly is the scenario we are imagining here? Would we have `.tif` files distributed over many folders? I assume we would want to generate spatial predictions for all of them?

I'm actually not working with `train.py` at the moment. My concrete use case was that I wanted to compute the embeddings for the inference patches. So I wanted to use `WorldCerealInferenceDataset` on its own but was stuck by the limitation of having it only look for files in the `data_dir`.

File: presto/dataset.py (hunk 1 above)
Comment (gabrieltseng) [reply presto/dataset.py:220]: does this work?:
```python
def __init__(self, input_dir: Path = data_dir / "inference_areas"):
```

Comment (kvantricht) [reply presto/dataset.py:220]: Apparently it does!

**Review by gabrieltseng — APPROVED** [review]

**Pull request merged**